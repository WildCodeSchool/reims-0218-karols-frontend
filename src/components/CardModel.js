import React from "react"
import { Card, CardTitle, CardBody, CardFooter } from "reactstrap"
import Info from "./Info"
import Counter from "./Counter"

const CardModel = ({
  id,
  index,
  title,
  description,
  image,
  selected,
  select,
  info,
  showCounter,
  count,
  handleMinus,
  handlePlus
}) => {
  return (
    <div>
      <Card
        style={{
          cursor: "pointer"
        }}
        onClick={() => {
          select(id)
        }}
      >
        <div
          style={{
            minHeight: "300px",
            position: "relative",
            overflow: "hidden",
            boxShadow:
              "0 19px 38px rgba(0, 0, 0, 0.3), 0 15px 12px rgba(0, 0, 0, 0.2)",
            background: selected
              ? `linear-gradient(
  rgba(255, 255, 255, 0.1), 
 rgba(255, 255, 255, 0.1)
 ), url(${image}) center/100% no-repeat`
              : `linear-gradient(
 rgba(0, 0, 0, 0.5), 
 rgba(0, 0, 0, 0.5)
 ), url(${image}) center/100% no-repeat`
          }}
        >
          <div
            className="data"
            style={{
              position: "absolute",
              bottom: "0",
              width: "100%",
              WebkitTransform: "translateY(calc(-10px + 1em))",
              transform: "translateY(calc(-10px + 1em))",
              transition: "-webkit-transform 0.3s"
            }}
          >
            <CardBody className="content">
              <CardTitle
                className="title"
                style={{
                  fontWeight: "500",
                  fontSize: "30px",
                  marginBottom: "120px",

                  backgroundColor: "rgba(255, 255, 255, 0.2)",
                  borderRadius: "10px",
                  width: "auto",
                  minWidth: "auto",
                  maxWidth: "auto",

                  background: selected
                    ? `linear-gradient(
                      rgba(0, 0, 0, 0.5), 
                      rgba(0, 0, 0, 0.5)
                      ),  center/100% no-repeat`
                    : `linear-gradient(
                      rgba(255, 255, 255, 0.1), 
                     rgba(255, 255, 255, 0.1)
                     ),  center/100% no-repeat`
                }}
              >
                <p
                  style={{
                    color: "#FFF"
                  }}
                >
                  {title}
                </p>
              </CardTitle>
            </CardBody>
            <CardFooter
              style={{
                backgroundColor: "rgba(136,136,136, 0.5)"
              }}
            >
              <p
                className="text"
                style={{
                  height: "70px",
                  margin: "0",
                  color: "#fff",
                  WebkitTransform: "translateY(calc(0px + 1em))",
                  transform: "translateY(calc(0px + 1em))"
                }}
              >
                {description}
              </p>
              {showCounter && (
                <Counter
                  count={count}
                  handleMinus={handleMinus}
                  handlePlus={handlePlus}
                />
              )}
              {info && <Info info={info} />}
              {/* 
              si info est true dans le json il renvoie le composant Info auquel on a pass√© la props info */}
            </CardFooter>
          </div>
        </div>
      </Card>
    </div>
  )
}

export default CardModel
