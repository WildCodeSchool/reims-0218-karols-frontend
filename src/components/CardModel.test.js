import React from "react"
import ReactDOM from "react-dom"
import CardModel from "./CardModel"
import renderer from "react-test-renderer"

it("renders without crashing", () => {
  const div = document.createElement("div")
  ReactDOM.render(<CardModel />, div)
  ReactDOM.unmountComponentAtNode(div)
})

it("renders correctly", () => {
  const tree = renderer.create(<CardModel />).toJSON()
  expect(tree).toMatchSnapshot()
})
