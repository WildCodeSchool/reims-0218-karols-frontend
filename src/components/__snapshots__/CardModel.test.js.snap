// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="card"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <div
      style={
        Object {
          "background": "linear-gradient(
         rgba(0, 0, 0, 0.5), 
         rgba(0, 0, 0, 0.5)
         ), url(undefined) center/100% no-repeat",
          "boxShadow": "0 19px 38px rgba(0, 0, 0, 0.3), 0 15px 12px rgba(0, 0, 0, 0.2)",
          "minHeight": "300px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        className="data"
        style={
          Object {
            "bottom": "0",
            "position": "absolute",
            "transform": "translateY(calc(-10px + 1em))",
            "transition": "-webkit-transform 0.3s, transform 0.3s, transform 0.3s, -webkit-transform 0.3s",
            "webkitTransform": "translateY(calc(-10px + 1em))",
            "width": "100%",
          }
        }
      >
        <div
          className="content card-body"
        >
          <h5
            className="title card-title"
            style={
              Object {
                "background": "linear-gradient(
                                    rgba(255, 255, 255, 0.1), 
                                   rgba(255, 255, 255, 0.1)
                                   ),  center/100% no-repeat",
                "backgroundColor": "rgba(255, 255, 255, 0.2)",
                "borderRadius": "10px",
                "fontSize": "30px",
                "fontWeight": "500",
                "marginBottom": "120px",
                "maxWidth": "auto",
                "minWidth": "auto",
                "width": "auto",
              }
            }
          >
            <p
              style={
                Object {
                  "color": "#FFF",
                }
              }
            />
          </h5>
        </div>
        <div
          className="card-footer"
          style={
            Object {
              "backgroundColor": "rgba(136,136,136, 0.5)",
            }
          }
        >
          <p
            className="text"
            style={
              Object {
                "color": "#fff",
                "height": "70px",
                "margin": "0",
                "transform": "translateY(calc(0px + 1em))",
                "webkitTransform": "translateY(calc(0px + 1em))",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
